guacamole-common-js/src/main/webapp/modules/BlobWriter.js

```diff
@@ -120,6 +124,12 @@ Guacamole.BlobWriter = function BlobWriter(stream) {
          */
         var readNextChunk = function readNextChunk() {
 
+            while (reader.readyState == 1) {
+                // The reader is busy so give it time to finish before proceeding.
+                // 1 second might be a bit generous, but it seems functional.
+                await sleep(1000);
```

Comment:
We cannot safely use `await` at this level of the API. Doing so would require dropping support for IE10+ and would require some significant discussion. I believe the same goes for `Promise` and the arrow notation.

That said, this may not matter: I don't believe we should use this approach. It may technically work, but there has to be a better way than polling `readyState` every second.

Are there no callbacks, events, etc. that can be used here such that we can rely on a call to `readNextChunk()` when the `FileReader` is actually ready?